rancher/charts
# [DNM] [rancher-monitoring] Fix Grafana links in local cluster

## Issue: rancher/rancher#41036 


## Problem

When some links are used to navigate around the Grafana dashboard, the UI breaks with an HTTP 404 error. After some investigation, it appears the UI is being served under `/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy/`, but our `nginx.conf` rules expect it to be `/k8s/clusters/local/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy/`.

I have not been able to figure out where the redirection from `/k8s/clusters/local/api/v1` to `/api/v1` comes from.

## Solution

Add a condition to serve Grafana from `/api/v1` in the local cluster, and from `/k8s/clusters/<cluster-id>/api/v1` on downstream clusters.

## Testing


## Engineering Testing
### Manual Testing

1. Install the Rancher Monitoring chart in a local cluster.
2. Navigate to the **Monitoring** tab.
3. Click **Grafana**.
4. Check the UI loads without showing an HTTP 404 cluster.
5. Repeat steps in a downstream cluster.

### Automated Testing

This chart does not have Hull testing yet.

## QA Testing Considerations


### Regressions Considerations


## Backporting considerations
